src/main/java/uk/gov/pay/connector/gateway/stripe/StripeGatewayClient.java

```
@@ -38,30 +40,23 @@ public StripeGatewayClient(Client client, MetricRegistry metricRegistry) {
         this.metricRegistry = metricRegistry;
     }
 
-    public Response postRequest(GatewayAccountEntity account,
-                                OrderRequestType requestType,
-                                URI url,
+    public Response postRequest(URI url,
                                 String payload,
-                                String authHeaderValue,
-                                MediaType mediaType) {
-        String metricsPrefix = format("gateway-operations.%s.%s.%s", account.getGatewayName(), account.getType(), requestType);
-
+                                Map<String, String> headers,
+                                MediaType mediaType,
+                                String metricsPrefix) throws GatewayClientException {
         Stopwatch responseTimeStopwatch = Stopwatch.createStarted();
         try {
-            logger.info("POSTing request for account '{}' with type '{}'", account.getGatewayName(), account.getType());
             Response response = client.target(url.toString())
                     .request()
-                    .header(AUTHORIZATION, authHeaderValue)
+                    .headers(new MultivaluedHashMap<>(headers))
                     .post(Entity.entity(payload, mediaType));
-
-            if (response.getStatusInfo().getFamily() == Response.Status.Family.SERVER_ERROR) {
-                logger.error("Stripe gateway returned server error: {}, for gateway url={} with type {}", response.getStatus(), url.toString(), account.getType());
-                incrementFailureCounter(metricRegistry, metricsPrefix);
-                throw new WebApplicationException("Unexpected HTTP status code " + response.getStatus() + " from gateway");
-            }
+            
+            throwIfErrorResponse(response, metricsPrefix);
```

👍 for throwing a GatewayClientException here but be careful as its currently swallowed by the `catch (Exception e)` below

---

ah, good spot